tests/unit/dashboard_preview/test_mode_preview.py

```diff
@@ -56,3 +68,138 @@ def test_get_preview_image_url_failure_missing_key(self) -> None:
+                'first_name': 'test_first_name',
+                'team_name': 'test_team',
+                'email': 'test_email',
+                'other_key_values': {}
+            }
+
+            preview = ModePreview(access_token='token', password='password', organization='foo')
+            self.assertRaises(PermissionError, preview._authorize_access, user_email='test_email')
+
+        with patch('amundsen_application.dashboard_preview.mode_preview.request_metadata') as mock_request_metadata:
+            mock_request_metadata.return_value.json.return_value = {
+                'employee_type': 'teamMember',
+                'full_name': 'test_full_name',
+                'is_active': 'True',
+                'github_username': 'test-github',
+                'slack_id': 'test_id',
+                'last_name': 'test_last_name',
+                'first_name': 'test_first_name',
+                'team_name': 'test_team',
+                'email': 'test_email',
+                'other_key_values': {
+                    'foo': 'bar'
+                }
+            }
+
+            preview = ModePreview(access_token='token', password='password', organization='foo')
+            self.assertRaises(PermissionError, preview._authorize_access, user_email='test_email')
+
+        with patch('amundsen_application.dashboard_preview.mode_preview.request_metadata') as mock_request_metadata:
+            mock_request_metadata.return_value.json.return_value = {
+                'employee_type': 'teamMember',
+                'full_name': 'test_full_name',
+                'is_active': 'True',
+                'github_username': 'test-github',
+                'slack_id': 'test_id',
+                'last_name': 'test_last_name',
+                'first_name': 'test_first_name',
+                'team_name': 'test_team',
+                'email': 'test_email',
+                'other_key_values': None
+            }
+
+            preview = ModePreview(access_token='token', password='password', organization='foo')
+            self.assertRaises(PermissionError, preview._authorize_access, user_email='test_email')
+
+        with patch('amundsen_application.dashboard_preview.mode_preview.request_metadata') as mock_request_metadata:
+            mock_request_metadata.return_value.json.return_value = {
+                'employee_type': 'teamMember',
+                'full_name': 'test_full_name',
+                'is_active': 'True',
+                'github_username': 'test-github',
+                'slack_id': 'test_id',
+                'last_name': 'test_last_name',
+                'first_name': 'test_first_name',
+                'team_name': 'test_team',
+                'email': 'test_email',
+            }
+
+            preview = ModePreview(access_token='token', password='password', organization='foo')
+            self.assertRaises(PermissionError, preview._authorize_access, user_email='test_email')
+
+
+if __name__ == '__main__':
```

Comment:
do you need this in order to get the unittests called?

Reply:
> do you need this in order to get the unittests called?

No, unit test itself will be executed vis pytest. But I found it that IDE that I am using - Pycharm - is not doing well on `debugging` when executing as `pytest`.